### DISCUSSION TITLE
Bar plots

### DISCUSSION THREAD

**Discussion by epagone** [kookma/ogpf#23] [Ideas]
Is there any ambition to add the ability to create bar plots?

**Comment by kookma** [comment]
Yes, there is! I welcome PRs and if I find time I will do myself. 

BUT for short solution/worraround by time ogpf gets its own function, please note that:

form ogpf you can issue any gnuplot command you wish! That means ogpf can do anything gnuplot can!
So, if there is a command not implemented in ogpf, simply generate data and call any plot!

Magic command: note that gp%option actually direct send any command to gnuplot!
also you can run scripts!

**Reply by TBurlacu-tud** [reply]
Do you think you could explain how I could use gp%option to call the splot command? For example, I have a 3D array that I would like to plot slices of.  

Thank you

**Reply by kookma** [reply]
* check the gnuplot site or open the example folder to see how gnuplot works for splot
* in Fortran you can create your data, save in a file
* in the same program you can use ogpf and directly call option
* each call to option can pass one command to gnuplot or you can use command delimiter to send more command
* that's all

**Reply by TBurlacu-tud** [reply]
I tried following [this example](http://www.gnuplotting.org/interpolation-of-heat-maps/) and creating a very simple file for plotting. 

Then I called as you suggested `call gp%options('plot "test.dat" matrix with image')`. However, nothing happens (no error message, no plot). 

I confirmed that calling the command from within gnuplot does work. Do you perhaps have a suggestion for how to proceed?

**Comment by kookma** [comment]
I think the path to the data file is not correct!

**Reply by TBurlacu-tud** [reply]
I tried with an absolute path as well and the result is the same. 

Moreover, a simple `call gp%options('plot sin(x)')` also does not work.

**Reply by kookma** [reply]
@TBurlacu-tud I will give a try at weekend and return to you!

**Reply by TBurlacu-tud** [reply]
Thank you!
